metricbeat/module/vsphere/host/host.go

```diff
@@ -240,3 +218,62 @@ func getAssetNames(ctx context.Context, pc *property.Collector, hs *mo.HostSyste
 		outputVmNames:      outputVmNames,
 	}, nil
 }
+
+func (m *HostMetricSet) getPerfMetrics(ctx context.Context, perfManager *performance.Manager, hst mo.HostSystem, metricIds []types.PerfMetricId) (map[string]interface{}, error) {
+	metricMap := make(map[string]interface{})
+	summary, err := perfManager.ProviderSummary(ctx, hst.Reference())
+	if err != nil {
+		return metricMap, fmt.Errorf("failed to get summary: %w", err)
+	}
+
+	var refreshRate = int32(m.Module().Config().Period.Seconds())
+	if summary.CurrentSupported {
+		refreshRate = summary.RefreshRate
+		if int32(m.Module().Config().Period.Seconds()) != refreshRate {
+			m.Logger().Warnf("User-provided period %v does not match system's refresh rate %v. Risk of data duplication. Consider adjusting period.", m.Module().Config().Period, refreshRate)
+		}
+	} else {
+		m.Logger().Warnf("Live data collection not supported. Use one of the system's historical interval (300, 1800, 7200, 86400). Risk of data duplication. Consider adjusting period.")
+	}
+
+	spec := types.PerfQuerySpec{
+		Entity:     hst.Reference(),
+		MetricId:   metricIds,
+		MaxSample:  1,
+		IntervalId: refreshRate,
+	}
+
+	// Query performance data
+	samples, err := perfManager.Query(ctx, []types.PerfQuerySpec{spec})
+	if err != nil {
+		return metricMap, fmt.Errorf("failed to query performance data: %w", err)
+	}
+
+	if len(samples) == 0 {
+		m.Logger().Debug("No samples returned from performance manager")
+		return metricMap, nil
+	}
+
+	results, err := perfManager.ToMetricSeries(ctx, samples)
+	if err != nil {
+		m.Logger().Errorf("failed to convert performance data to metric series: %v", err)
```

Comment:
Is it okay if we continue with the function execution if there's an error here? Should we consider returning an error instead of just logging?

Reply:
Yes, we need to continue because there are at least other metrics that we are getting.
We can skip assets.